### PR TITLE
ESDT improvements new functions

### PR DESCRIPTION
**Pull request by ssd04**
#### Did you test the changes locally ?
- [ ] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [ ] document new feature
- [ ] update documentation that is not relevant anymore
- [ ] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/tokens/nft-tokens.mdx [hunk 1]
```diff
@@ -698,6 +703,120 @@ _For more details about how arguments have to be encoded, check [here](/develope
...
+}
+```
+
+_For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._
+
+[comment]: # (mx-context-auto)
+
+### **MetaData Update**
+
+The NFT meta data can be updated using a transaction like this:
```
Comment (BeniaminDrasovean) [inline docs/tokens/nft-tokens.mdx:772]: Also add: `If nothing is received for a given attribute, the old version of that attribute will be kept`

File: docs/tokens/nft-tokens.mdx [hunk 2]
```diff
@@ -698,6 +703,120 @@ _For more details about how arguments have to be encoded, check [here](/develope
...
+}
+```
+
+_For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._
+
+[comment]: # (mx-context-auto)
+
+### **MetaData Recreate**
+
+The whole NFT can be recreated with new attributes using a transaction like this:
```
Comment (BeniaminDrasovean) [inline docs/tokens/nft-tokens.mdx:797]: Also add: `If an argument is not received, that field is set to zero`

File: docs/tokens/nft-tokens.mdx [hunk 3]
```diff
@@ -698,6 +703,120 @@ _For more details about how arguments have to be encoded, check [here](/develope
...
+}
+```
+
+_For more details about how arguments have to be encoded, check [here](/developers/sc-calls-format)._
+
+[comment]: # (mx-context-auto)
+
+### **Set new URIs**
+
+The manager of an ESDT token may want to set new uris. This operation will rewrite the uris on the specified token ID. It requires `ESDTRoleNFTSetNewURIs` role.
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply docs/tokens/nft-tokens.mdx:730]: For both appearances

[the remark this answers is not in the document]
Comment (ssd04) [reply docs/tokens/nft-tokens.mdx]: done

File: docs/tokens/nft-tokens.mdx (hunk 1 above)
Comment (ssd04) [reply docs/tokens/nft-tokens.mdx]: done

File: docs/tokens/nft-tokens.mdx (hunk 2 above)
Comment (ssd04) [reply docs/tokens/nft-tokens.mdx:814]: done

**Review by sasurobert — DISMISSED** [review]

**Review by BeniaminDrasovean — DISMISSED** [review]

**Review by BeniaminDrasovean — DISMISSED** [review]

**Review by andreibancioiu — DISMISSED** [review]

File: docs/tokens/nft-tokens.mdx [hunk 4]
```diff
@@ -293,13 +293,23 @@ For NFT:
...
+- ESDTRoleModifyCreator : this role allows one to rewrite the creator of a specific token
+- ESDTRoleNFTRecreate : this role allows one to recreate the whole NFT with new attributes
 
 For SFT:
 
 - ESDTRoleNFTCreate : this role allows one to create a new SFT
 - ESDTRoleNFTBurn : this role allows one to burn quantity of a specific SFT
 - ESDTRoleNFTAddQuantity : this role allows one to add quantity of a specific SFT
 - ESDTTransferRole : this role enables transfer only to specified addresses. The addresses with the transfer role can transfer anywhere.
+- ESDTRoleNFTUpdate : this role allows one to update meta data attributes of a specific NFT
```
Comment (Jotenks66) [inline docs/tokens/nft-tokens.mdx:308]: "of a specific SFT" instead of NFT to stay consistent with the rest.

Comment (ssd04) [reply docs/tokens/nft-tokens.mdx]: updated, thanks

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**